fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -2015,12 +2077,15 @@ struct RestoreClusterImpl {
 		// Mark tenants that are missing from the data cluster in an error state on the management cluster
 		wait(processMissingTenants(self));
 
-		// set restored cluster to ready state
 		if (!self->restoreDryRun) {
-			wait(self->ctx.runManagementTransaction([self = self](Reference<typename DB::TransactionT> tr) {
-				self->markClusterAsReady(tr);
-				return Future<Void>(Void());
-			}));
+			// Remove the active restore ID from the data cluster
+			wait(self->ctx.runDataClusterTransaction(
+			    [self = self](Reference<ITransaction> tr) { return eraseRestoreId(self, tr); }));
+
+			// set restored cluster to ready state
+			wait(self->ctx.runManagementTransaction(
+			    [self = self](Reference<typename DB::TransactionT> tr) { return self->markClusterAsReady(tr); }));
+			TraceEvent("MarkedDataClusterReady").detail("Name", self->clusterName);
```

Comment:
This should probably be named differently than line 2140 so there is no ambiguity about its source

Reply:
Or turned into a `CODE_PROBE` if this doesn't need to be a trace event

Reply:
Yeah, I can do that. This trace event was extracted from the `markClusterAsReady` function called in both locations, so it isn't really a change from before, but since it's now easier to write a distinct trace event for the two cases I can do so.